lib/register.coffee

```diff
@@ -96,7 +97,11 @@ module.exports = ({ request }) ->
 				api_key: options.deviceApiKey
 		.tap (response) ->
 			if response.statusCode isnt 201
-				throw new Error(response.body)
+				throw new ApiError(response.body, response)
 		.get('body')
 		# Allow promise based and callback based styles
 		.asCallback(callback)
+
+class ApiError extends TypedError
+	constructor: (message = 'Error with API request', @response) ->
+		super(@message)
```

Comment:
```suggestion
	constructor: (message = 'Error with API request', @response) ->
		super(message)
		@message = message
```
That's the simplest way to get this emit a valid build result, otherwise we can't pass `@message` to `super()`, since `this` is undefined at that point.